### ISSUE TITLE
Use a scheduler instead of ForkJoinPool on JVM?

### ISSUE THREAD

**Issue by He-Pin** [lampepfl/gears#77]
Motivation:
As in PR https://github.com/com-lihaoyi/cask/pull/125/commits/17dbf69d43bb4027c14caf2c7fc1140b9b5a9f50 I found it's possible to run the virtual thread on another thread pool instead of the default scheduler.

I think that maybe a little lightweight in some cases if the task never forks.
what do you think, I would like to contribute this if you like.

**Comment by natsukagami** [comment]
Yes, I think this is definitely useful! Perhaps custom thread pools can be implemented as an instance of `Scheduler`

**Comment by He-Pin** [comment]
Yes, that's the whole point, I will prepare it after work today:)
